docs/observability/how_to_guides/tracing/mask_inputs_outputs.mdx

```diff
@@ -292,3 +292,317 @@ async def async_function(key: str) -> int:
+            anonymized_list.append(anonymized_item)
+        return anonymized_list
+    elif isinstance(data, str):
+        entities = detect_pii(data)
+        anonymized_data = redact_pii_entities(data, entities)
+        return anonymized_data
+    else:
+        return data
+
+# wrap the openai client
+openai_client = wrap_openai(openai.Client())
+
+# initialize the langsmith client with the anonymization functions
+langsmith_client = Client(
+  hide_inputs=recursive_anonymize, hide_outputs=recursive_anonymize
+)
+
+# The trace produced will have its metadata present, but the inputs and outputs will be anonymized
+response_with_anonymization = openai_client.chat.completions.create(
+  model="gpt-4o-mini",
+  messages=[
+      {"role": "system", "content": "You are a helpful assistant."},
+      {"role": "user", "content": "My name is Slim Shady, call me at 313-666-7440 or email me at real.slim.shady@gmail.com"},
+  ],
+  langsmith_extra={"client": langsmith_client},
+)
+
+# The trace produced will not have anonymized inputs and outputs
+response_without_anonymization = openai_client.chat.completions.create(
+  model="gpt-4o-mini",
+  messages=[
+      {"role": "system", "content": "You are a helpful assistant."},
+      {"role": "user", "content": "My name is Slim Shady, call me at 313-666-7440 or email me at real.slim.shady@gmail.com"},
+  ],
+)
+```
+
+The anonymized run will look like this in LangSmith:
+![Anonymized run](./static/aws-comprehend-anonymized.png)
+
+The non-anonymized run will look like this in LangSmith:
+![Non-anonymized run](./static/aws-comprehend-not-anonymized.png)
+
+### Regex
+
+If you would prefer to not use an external anonymizer, you can use regex to mask inputs and outputs. The implementation below uses regex to mask email addresses, phone numbers, full names, credit card numbers, and SSNs.
+
+```python
+import re
+import openai
+from langsmith import Client
+from langsmith.wrappers import wrap_openai
+
+# Define regex patterns for various PII
+SSN_PATTERN = re.compile(r'\b\d{3}-\d{2}-\d{4}\b')
+CREDIT_CARD_PATTERN = re.compile(r'\b(?:\d[ -]*?){13,16}\b')
+EMAIL_PATTERN = re.compile(r'\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,7}\b')
+PHONE_PATTERN = re.compile(r'\b(?:\+?1[-.\s]?)?\(?\d{3}\)?[-.\s]?\d{3}[-.\s]?\d{4}\b')
+FULL_NAME_PATTERN = re.compile(r'\b([A-Z][a-z]*\s[A-Z][a-z]*)\b')
+
+def regex_anonymize(text):
+    """
+    Anonymize sensitive information in the text using regex patterns.
+
+    Args:
+        text (str): The input text to be anonymized.
+
+    Returns:
+        str: The anonymized text.
+    """
+    # Replace SSNs with a placeholder
+    text = SSN_PATTERN.sub('[REDACTED SSN]', text)
+    # Replace credit card numbers with a placeholder
+    text = CREDIT_CARD_PATTERN.sub('[REDACTED CREDIT CARD]', text)
+    # Replace email addresses with a placeholder
+    text = EMAIL_PATTERN.sub('[REDACTED EMAIL]', text)
+    # Replace phone numbers with a placeholder
```

Comment:
IDK how useful these redundant comments are tbh

Reply:
yep agree. just added a general one at the top + deleted the rest